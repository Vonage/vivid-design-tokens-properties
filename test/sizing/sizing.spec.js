import sizingTokens from '../../globals/sizing/index.js';
import { rawData, expectedResult } from './test-data.js';
import sizingParser from '../../scripts/figma-parsers/parsers/sizing-figma-data-parser.js';

describe(`Sizing Tokens`, function () {
    it(`should create the sizing tokens`, function () {
        expect(JSON.stringify(sizingTokens, null, 2)).toMatchSnapshot();
    });

    describe(`Figma Parser`, function () {
        let results = {};

        function mockWriteJson(data, path) {
            results.path = path;
            results.data = data;
        }

        beforeEach(function () {
            results = {};
        });

        describe(`Sizing`, function () {
            it(`should write to JSON`, function () {
                sizingParser.parse(rawData, mockWriteJson);
                expect(Object.keys(results.data).length).toEqual(1);
            });

            it(`should generate the correct sizing values json from Figma`, function () {
                sizingParser.parse(rawData, mockWriteJson);
                expect({...expectedResult}).toEqual(results.data);
            });

            it(`should save to correct files paths`, function () {
                sizingParser.parse(rawData, mockWriteJson);
                expect(results.path).toEqual(`./dist/sizing/sizing.values.json`);
            });
        });
    });
});
