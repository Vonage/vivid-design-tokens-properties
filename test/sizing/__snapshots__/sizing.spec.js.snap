// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sizing Tokens should create the sizing tokens 1`] = `
"{
  \\"sizing\\": {
    \\"4px-scale\\": {
      \\"3xs\\": {
        \\"value\\": \\"{alias.sizing.4px-scale.3xs.value}\\"
      },
      \\"2xs\\": {
        \\"value\\": \\"{alias.sizing.4px-scale.2xs.value}\\"
      },
      \\"xs\\": {
        \\"value\\": \\"{alias.sizing.4px-scale.xs.value}\\"
      },
      \\"sm\\": {
        \\"value\\": \\"{alias.sizing.4px-scale.sm.value}\\"
      },
      \\"md\\": {
        \\"value\\": \\"{alias.sizing.4px-scale.md.value}\\"
      },
      \\"lg\\": {
        \\"value\\": \\"{alias.sizing.4px-scale.lg.value}\\"
      },
      \\"xl\\": {
        \\"value\\": \\"{alias.sizing.4px-scale.xl.value}\\"
      },
      \\"2xl\\": {
        \\"value\\": \\"{alias.sizing.4px-scale.2xl.value}\\"
      },
      \\"3xl\\": {
        \\"value\\": \\"{alias.sizing.4px-scale.3xl.value}\\"
      }
    },
    \\"8px-scale\\": {
      \\"3xs\\": {
        \\"value\\": \\"{alias.sizing.8px-scale.3xs.value}\\"
      },
      \\"2xs\\": {
        \\"value\\": \\"{alias.sizing.8px-scale.2xs.value}\\"
      },
      \\"xs\\": {
        \\"value\\": \\"{alias.sizing.8px-scale.xs.value}\\"
      },
      \\"sm\\": {
        \\"value\\": \\"{alias.sizing.8px-scale.sm.value}\\"
      },
      \\"md\\": {
        \\"value\\": \\"{alias.sizing.8px-scale.md.value}\\"
      },
      \\"lg\\": {
        \\"value\\": \\"{alias.sizing.8px-scale.lg.value}\\"
      },
      \\"xl\\": {
        \\"value\\": \\"{alias.sizing.8px-scale.xl.value}\\"
      },
      \\"2xl\\": {
        \\"value\\": \\"{alias.sizing.8px-scale.2xl.value}\\"
      },
      \\"3xl\\": {
        \\"value\\": \\"{alias.sizing.8px-scale.3xl.value}\\"
      }
    }
  }
}"
`;
